quickwit-indexing/src/actors/pipeline_supervisor.rs

```diff
@@ -405,16 +407,41 @@ impl IndexingPipelineSupervisor {
         Ok(())
     }
 
+    async fn process_spawn(
+        &mut self,
+        ctx: &ActorContext<Msg>,
+        retry_count: usize,
+    ) -> Result<(), ActorExitStatus> {
+        match retry_count {
+            0 | 1 | 2 => {
+                let duration = if retry_count == 0 {
+                    Duration::from_millis(1000)
+                } else if retry_count == 1 {
+                    Duration::from_millis(2000)
+                } else {
+                    Duration::from_millis(5000)
+                };
+
+                if let Err(spawn_error) = self.spawn_pipeline(ctx).await {
+                    warn!(err=?spawn_error, "Error while spawn_pipeline, waiting 1000ms");
+                    self.terminate().await;
+                    ctx.schedule_self_msg_blocking(
+                        duration,
+                        Msg::Spawn {
+                            retry_count: retry_count + 1,
+                        },
+                    );
+                }
+            }
+            _ => {
+                self.spawn_pipeline(ctx).await?;
```

Comment:
I think actually we want to keep retrying forever, at least for the moment.

Soooo:
- retry forever
- exponential backoff until the retry goes up to say 5mn, after which we stagnate to 5mn. (It actually simpliy the code too)

Reply:
if it's just for now, I would keep the logic like it is and just always return Some(duration)

Reply:
Makes sense.